api_tests/registrations/views/test_withdrawn_registrations.py

```diff
@@ -0,0 +1,173 @@
+        self.registration.reload()
+        assert_equal(res.status_code, 405)
+        assert_equal(self.registration.title, self.registration.title)
+        assert_equal(self.registration.description, self.registration.description)
+
+    def test_cannot_access_withdrawn_files_list(self):
+        url = '/{}registrations/{}/files/'.format(API_BASE, self.registration._id)
+        res = self.app.get(url, auth=self.user.auth, expect_errors=True)
+        assert_equal(res.status_code, 403)
+
+    def test_cannot_access_withdrawn_node_links_detail(self):
+        url = '/{}registrations/{}/node_links/{}/'.format(API_BASE, self.registration._id, self.public_pointer._id)
+        res = self.app.get(url, auth=self.user.auth, expect_errors=True)
+        assert_equal(res.status_code, 403)
+
+    def test_cannot_access_withdrawn_node_links_list(self):
+        url = '/{}registrations/{}/node_links/'.format(API_BASE, self.registration._id)
+        res = self.app.get(url, auth=self.user.auth, expect_errors=True)
+        assert_equal(res.status_code, 403)
+
+    def test_cannot_access_withdrawn_node_logs(self):
+        self.public_project = ProjectFactory(is_public=True, creator=self.user)
+        url = '/{}registrations/{}/logs/'.format(API_BASE, self.registration._id)
+        res = self.app.get(url, auth=self.user.auth, expect_errors=True)
+        assert_equal(res.status_code, 403)
+
+    def test_cannot_access_withdrawn_registrations_list(self):
+        self.registration.save()
+        url = '/{}registrations/{}/registrations/'.format(API_BASE, self.registration._id)
+        res = self.app.get(url, auth=self.user.auth, expect_errors=True)
+        assert_equal(res.status_code, 403)
+
+    def test_withdrawn_registrations_display_limited_fields(self):
+        registration = self.registration
+        res = self.app.get(self.withdrawn_url, auth=self.user.auth)
+        assert_equal(res.status_code, 200)
+        attributes = res.json['data']['attributes']
+        expected_attributes = {
+            'title': registration.title,
+            'description': registration.description,
+            'date_created': registration.date_created.isoformat(),
+            'date_registered': registration.registered_date.isoformat(),
+            'withdrawal_justification': registration.retraction.justification,
+            'public': None,
+            'category': None,
+            'date_modified': None,
+            'registration': True,
+            'fork': None,
+            'collection': None,
+            'tags': None,
+            'withdrawn': True,
+            'pending_withdrawal': None,
+            'pending_registration_approval': None,
+            'pending_embargo_approval': None,
+            'embargo_end_date': None,
+            'registered_meta': None,
+            'current_user_permissions': None,
+            'registration_supplement': registration.registered_schema[0].name
+        }
+
+        for attribute in expected_attributes:
```

Comment:
The big gap in testing is that, if someone adds a new field to the NodeSerializer but doesn't mark it explicitly in the RegistrationSerializer, then it's going to be serialized. That will be very easy to miss. Can you either add a new test or refactor this test to account for that? It might make more sense as a Serializer test. 

The other option is to refactor the hidden fields to work more like the Anonymous fields, where we have to explicitly list the fields that are okay to show. Mostly, though, just make sure people won't accidentally reveal information by adding a new field to the Node serializer. Probably we'll need to do something similar with the relationships fields.